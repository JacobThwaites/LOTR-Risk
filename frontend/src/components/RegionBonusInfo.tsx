import React from 'react';
import { Regions } from '../gameLogic/Enums/Regions';

export default function RegionBonusInfo() {
    return (
        <div className='region-bonus-info'>
            <h1 className='region-bonus-info--header'>Region Bonuses</h1>
            {Object.keys(Regions).map((keyName, i) => (
                <p className={`region-bonus-info--region ${Regions[keyName].getName()}`} key={i}>{Regions[keyName].getName()} +{Regions[keyName].getBonusUnits()}</p>
            ))}
        </div>
    )
}