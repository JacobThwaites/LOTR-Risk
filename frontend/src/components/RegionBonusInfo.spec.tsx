import '@testing-library/jest-dom'
import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react'
import RegionBonusInfo from './RegionBonusInfo';


it('renders RegionBonusInfo correctly', () => {
    const component = renderer.create(
        <RegionBonusInfo />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('show info for all regions', () => {
    const { container } = render(<RegionBonusInfo />)
    expect(container.getElementsByClassName('region-bonus-info--region').length).toBe(9);
});
