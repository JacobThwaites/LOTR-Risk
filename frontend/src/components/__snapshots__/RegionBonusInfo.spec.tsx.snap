// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RegionBonusInfo correctly 1`] = `
<div
  className="region-bonus-info"
>
  <h1
    className="region-bonus-info--header"
  >
    Region Bonuses
  </h1>
  <p
    className="region-bonus-info--region eriador"
  >
    eriador
     +
    3
  </p>
  <p
    className="region-bonus-info--region arnor"
  >
    arnor
     +
    7
  </p>
  <p
    className="region-bonus-info--region rhun"
  >
    rhun
     +
    2
  </p>
  <p
    className="region-bonus-info--region mirkwood"
  >
    mirkwood
     +
    4
  </p>
  <p
    className="region-bonus-info--region rohan"
  >
    rohan
     +
    4
  </p>
  <p
    className="region-bonus-info--region rhovanion"
  >
    rhovanion
     +
    4
  </p>
  <p
    className="region-bonus-info--region gondor"
  >
    gondor
     +
    7
  </p>
  <p
    className="region-bonus-info--region mordor"
  >
    mordor
     +
    2
  </p>
  <p
    className="region-bonus-info--region haradwaith"
  >
    haradwaith
     +
    2
  </p>
</div>
`;
